File: src/main/java/InvertBinaryTree/zhanwentao/InvertBinaryTree.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
#!@Author:zhangwentao


"""
left -> right
right -> left
"""
class Solution():
    def invertTree(self, root):
        if root == None: return root
        temp = root.left
        root.left = root.right
        root.right = temp
        self.invertTree(root.left)
        self.invertTree(root.right)
        return root

File: src/main/java/InvertBinaryTree/lion/InvertBinaryTree.java
package InvertBinaryTree.lion;

public class InvertBinaryTree {
	public TreeNode invertTree(TreeNode root) {
		if (root == null) {
			return null;
		}
		TreeNode temp = root.left;
		root.left = root.right;
		root.right = temp;
		invertTree(root.left);
		invertTree(root.right);
		return root;
	}
}

class TreeNode {
	int val;
	TreeNode left;
	TreeNode right;

	TreeNode(int x) {
		val = x;
	}
}
